utils/s2n_mem.c

```diff
@@ -73,10 +73,9 @@ static int s2n_mem_cleanup_impl(void)
 
 static int s2n_mem_free_mlock_impl(void *ptr, uint32_t size)
 {
-    int munlock_rc = munlock(ptr, size);
+    /* Perform a best-effort `munlock`: ignore any errors during unlocking. */
+    munlock(ptr, size);
     free(ptr);
-    POSIX_ENSURE(munlock_rc == 0, S2N_ERR_MUNLOCK);
```

Comment:
should we also remove `S2N_ERR_MUNLOCK` from s2n_errno.c since this doesnt exist anymore?

Reply:
this might be a breaking api change so possibly no good